src/Stratis.Bitcoin/P2P/PeerSelector.cs

```diff
@@ -323,7 +323,7 @@ public IEnumerable<PeerAddress> Attempted()
         {
             return this.peerAddresses.Values.Where(p =>
                 p.Attempted &&
-                p.ConnectionAttempts < PeerAddress.AttemptThreshold &&
+                p.ConnectionAttempts <= PeerAddress.AttemptThreshold &&
```

Comment:
It seems that if the peer has been attempted precisely `AttemptThreshold` times then you have an overlap between getting returned in the `Attempted()` list and getting regarded as a candidate for `CanResetAttempts`. Is this intentional?

Reply:
@zeptin your analysis is 100% correct. The issue here was there we weren't returning the peer to be reset. I'm not too happy with the design here but I dont wanna go through a refactor now :P